pkg/apimachinery/utils/provisioning.go

```diff
@@ -0,0 +1,58 @@
+package utils
+
+import "time"
+
+// ManagerProperties is used to identify the manager of the resource.
+//
+// This is used to identify the manager of the resource.
+//
+// The manager kind is the type of manager, such as "ui", "api/generic", "api/kubectl", or "api/terraform".
+//
+// The manager identity is the identity of the manager, such as the username of the user who created the resource,
+// or the name of the tool that created the resource.
+//
+// The is exclusive flag indicates whether the manager is the exclusive owner of the resource.
+// If set to true, then only updates coming from the manager will be accepted.
+type ManagerProperties struct {
+	Kind        ManagerKind
+	Identity    string
+	AllowsEdits bool
+	Suspended   bool
+}
+
+// ManagerKind is the type of manager, which is responsible for managing the resource.
+// It can be a user or a tool or a generic API client.
+type ManagerKind string
+
+// Known values for ManagerKind.
+const (
+	ManagerKindUnknown   ManagerKind = ""
```

Comment:
does this exist?  the same as not having an annotation right?

Reply:
Yes, this is the fallback value when nothing is set.